feat(grid): implement column resizing

trajchen opened the pull request.

trajchenaumoski committed:
feat(grid): implement column resizing

trajchenaumoski committed:
feat(grid): add a selector that will return all grid rows

velinovskav commented on projects/data-grid/src/lib/directives/column-resize.directive.ts:
```diff
@@ -0,0 +1,99 @@
+  }
+
+  private resizeColumn(currentPositionX: number): DataGridColumnWithId {
+    return R.mergeRight(this.column, {width: this.calculateColumnWidth(currentPositionX)});
```
Why are we changing the width of the column here? Shouldn't we just calculate the new width and update the width of the column in reducer.

velinovskav commented on projects/data-grid/src/lib/directives/column-resize.directive.ts:
```diff
@@ -0,0 +1,99 @@
+  @Input() column: DataGridColumnWithId;
+
+  @Output() readonly columnResizing = new EventEmitter<DataGridColumnWithId>();
+  @Output() readonly columnResized = new EventEmitter<DataGridColumnWithId>();
```
IMO these events should keep affected column and new width.

trajchenaumoski committed:
feat(grid): add a model for the column resize events

trajchenaumoski committed:
feat(grid): increase the children paddings

trajchenaumoski committed:
feat(grid): replace the event enum with string

Replace the event enum with string because it is not allowed to be used in the host listener.

trajchenaumoski committed:
feat(grid): add initial column width in resize directive
